### PR TITLE
doc sign out app

### PR DESCRIPTION
**Pull request by akanex115**
# Background
https://github.com/ajktown/docs/issues/41

## TODOs
- [x] Document onSignOutApp() of src/hooks/�app/use-on-sign-out-app.hook.ts

## Related PRs
- _None_

## Checklist (Developers)
- [x] Make the PR as the draft
- [x] Title is checked
- [x] Background & TODOs are filled
- [x] `Related PRs` is correctly modified, if it is not `None`
- [x] Assignee is set
- [x] Labels are set
- [x] Appropriate issue(s) is(are) linked to this PR under `development`
- [x] `yarn inspect` is run
- [x] `TODOs` are handled and checked
- [x] Final Operation Check is done

## Checklist (Reviewers)
- [x] Check if there are any other missing TODOs that are not yet listed
- [x] Review Code
- [x] Every item on the checklist has been addressed accordingly
- [x] If `development` is associated to this PR, you must check if every TODOs are handled

### CODE REVIEW

File: dev/diagrams/on-sign-out-app-hook.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+box "API" #Lightgreen
+  participant "API" as api
+box
+
+activate user
+  user -> fe: SignOutApp
+  activate fe
+    fe -> hook: onSignOutAppHook
+    activate hook
+      hook -> api: postSignOut
```
Comment (mlajkim) [inline dev/diagrams/on-sign-out-app-hook.md:36]: image
`await`をするってことは、ちゃんとAPI交信が終わるまで待つ、ということなので

image

「赤」ではなく「オレンジ」に配置しないといけないね

File: dev/diagrams/on-sign-out-app-hook.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
+# On Sign Out App Hook
+
```
Comment (mlajkim) [inline dev/diagrams/on-sign-out-app-hook.md:2]: そもそもfile 名 `dev/diagrams/on-sign-out-app-hook.md`よりは、
`dev/diagrams/on-sign-out-app.md`かな？

File: dev/diagrams/on-sign-out-app-hook.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
+# On Sign Out App Hook
```
Comment (mlajkim) [inline dev/diagrams/on-sign-out-app-hook.md:1]:
```suggestion
# On Sign Out App
```

**Review by mlajkim — COMMENTED** [review]
first review done.
全体的にさらによく書けるようになった気がするね。

File: dev/diagrams/on-sign-out-app-hook.md (hunk 2 above)
Comment (akanex115) [reply dev/diagrams/on-sign-out-app-hook.md]: Handled in [9963c14](https://github.com/ajktown/docs/pull/47/commits/9963c14e47fee57e7b97a0c6c8dd979ab28ae6f8)

File: dev/diagrams/on-sign-out-app-hook.md (hunk 1 above)
Comment (akanex115) [reply dev/diagrams/on-sign-out-app-hook.md]: Handled in [e2e8e9a](https://github.com/ajktown/docs/pull/47/commits/e2e8e9ae77ff12f833f2e99660016a9ae122045c)

File: dev/diagrams/on-sign-out-app.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+      hook -> api: postSignOut
+      activate api
+        hook <- api: Sign-out successful
+      deactivate api
+      hook -> recoil: Resets wordIdsState
+      activate recoil
+        hook -> recoil: Resets preferenceState
+        recoil -> recoil: Resets Recoil states
+        hook <- recoil: Return nothing
+      deactivate recoil
```
Comment (mlajkim) [inline dev/diagrams/on-sign-out-app.md:45]:
```suggestion
      hook -> recoil: Resets wordIdsState
      hook -> recoil: Resets preferenceState
```

これで十分かな？

File: dev/diagrams/on-sign-out-app.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+@startuml
+
+title onSignOutApp()
+
+participant "End User" as user
+
+box "FE" #Lightblue
+  participant "Component" as fe
+  participant "Hook: onSignOutApp" as hook
```
Comment (mlajkim) [inline dev/diagrams/on-sign-out-app.md:24]:
```suggestion
  participant "Hook: useSignOutApp" as hook
```

**Review by mlajkim — COMMENTED** [review]
Review done

**Review by mlajkim — APPROVED** [review]
OK

**Pull request merged**